### ISSUE TITLE
Bug: When setting asFormData to true null values will be sent as empty string

### ISSUE THREAD

**Issue by SpaceOgre** [FastEndpoints/FastEndpoints#850]
When using the integration test methods and setting the parameter `asFormData ` to `true` the properties marked as nullable will be sent as empty string instead of omitted, causing the server to set default values instead of null (at least for primitv values like Guid).

The fix you introduce here: https://github.com/FastEndpoints/FastEndpoints/blob/e229ff9c716ab8a457a74a724569a421753f03df/Src/Library/Testing/HttpClientExtensions.cs#L427C17-L427C39

Fixes this issue.

But the newly introduced Json Seriliztion of the values for the StringContent cause me to get this error:
```
{"name":"legalProvisionId","reason":"Value [\"8267237c-a8f4-4882-bc05-4e1b7d665093\"] is not valid for a [Guid] property!"}
```
I do not get this error when just using ToString on the value.

To recreate this error you can modify one of your tests:
```c#
    [Fact]
    public async Task FileHandlingFileBinding()
    {
        for (var i = 0; i < 3; i++) //repeat upload multiple times
        {
            await using var stream1 = File.OpenRead("test.png");
            await using var stream2 = File.OpenRead("test.png");
            await using var stream3 = File.OpenRead("test.png");

            var req = new Uploads.Image.SaveTyped.Request
            {
                File1 = new FormFile(stream1, 0, stream1.Length, "File1", "test.png")
                {
                    Headers = new HeaderDictionary(),
                    ContentType = "image/png"
                },
                File2 = new FormFile(stream2, 0, stream2.Length, "File2", "test.png"),
                File3 = new FormFile(stream3, 0, stream2.Length, "File3", "test.png"),
                Width = 500,
                Height = 500,
                GuidId = Guid.NewGuid(), //Add this
            };

            var res = await App.AdminClient.POSTAsync<
                          Uploads.Image.SaveTyped.Endpoint,
                          Uploads.Image.SaveTyped.Request>(req, sendAsFormData: true);

            using var md5Instance = MD5.Create();
            await using var stream = await res.Content.ReadAsStreamAsync();
            var resMd5 = BitConverter.ToString(md5Instance.ComputeHash(stream)).Replace("-", "");
            var resJson = await res.Content.ReadAsStringAsync(); //Add this to easily see the result error

            resMd5.Should().Be("8A1F6A8E27D2E440280050DA549CBE3E");
        }
    }
```
And just add the `GuidId` property to the Request class.

Then we get this error:
`{"StatusCode":400,"Message":"One or more errors occurred!","Errors":{"GuidId":["Value [\"4610ede6-e4a4-4747-8a91-b31071708466\"] is not valid for a [Guid] property!"]}}`

**Comment by dj-nitehawk** [comment]
fixed in `v5.32.0.8-beta`
give that a go and report back pls.

**Comment by SpaceOgre** [comment]
This fix is working nicely for me 👍
